compiler/prelude/prelude.go

```diff
@@ -120,7 +120,7 @@ var $substring = function(str, low, high) {
 
 var $sliceToArray = function(slice) {
   if (slice.$length === 0) {
-    return [];
+    return new slice.$array.constructor;
```

Comment:
I tested with that entirely removed at first, and it worked just as well. I wondered if the special case served as a meaningful optimization, so left it in there.

If you're in favor of removing it, that is of course fine with me. I can make that change.